06.bowling_object/frame.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Frame
+  attr_reader :shots
+
+  def initialize(index, shots)
+    @index = index
+    @shots = shots
+  end
+
+  def calculate_score(frames)
+    score_without_bonus + calculate_bonus(frames)
+  end
+
+  private
+
+  def score_without_bonus
+    @shots.sum(&:score)
+  end
+
+  def strike?
+    @shots[0].strike?
+  end
+
+  def spare?
+    @shots.first(2).sum(&:score) == 10 && !strike?
+  end
+
+  def calculate_bonus(frames)
+    return 0 if @index >= 9
+
+    next_frame = frames[@index + 1]
+    after_next_frame = frames[@index + 2]
+    if strike?
+      (next_frame.shots + (after_next_frame&.shots || [])).first(2).sum(&:score)
```

Comment:
```suggestion
      following_shots = next_frame.shots + (after_next_frame&.shots || [])
      following_shots.first(2).sum(&:score)
```

いったん変数に入れた方が読みやすいと思いました。
https://x.com/jnchito/status/1326468865103073280

その上で、after_next_frameの条件分岐も行を分けるとさらに読み手に優しいコードになるかもしれません。（ここは好みの問題ですが）

```suggestion
      following_shots = next_frame.shots
      following_shots += after_next_frame.shots if after_next_frame
      following_shots.first(2).sum(&:score)
```